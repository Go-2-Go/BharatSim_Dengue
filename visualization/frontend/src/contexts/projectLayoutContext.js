import React from 'react';

const projectLayoutContext = React.createContext({
  projectMetadata: {
    name:'',
  },
  selectedDashboardMetadata: {
    name: '',
  },
});

const ProjectLayoutProvider = projectLayoutContext.Provider;

export { projectLayoutContext, ProjectLayoutProvider };
