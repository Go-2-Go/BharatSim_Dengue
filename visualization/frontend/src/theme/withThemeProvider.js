import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme() {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
