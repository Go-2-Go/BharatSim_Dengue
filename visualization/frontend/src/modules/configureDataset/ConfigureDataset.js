import React, { useContext } from 'react';
import Box from '@material-ui/core/Box';
import { Typography } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import { useHistory, Link as RouterLink } from 'react-router-dom';
import Link from '@material-ui/core/Link';
import { projectLayoutContext } from '../../contexts/projectLayoutContext';
import useConfigureDatasetStyles from './configureDatasetCSS';

function ConfigureDataset() {
  const classes = useConfigureDatasetStyles();
  const history = useHistory();

  const { projectMetadata, selectedDashboardMetadata } = useContext(projectLayoutContext);

  function openRecentProjects() {
    history.push('/');
  }

  const uploadFilePage = `/projects/${projectMetadata.id}/upload-dataset`;

  return (
    <Box>
      <Box className={classes.configureProjectDataBar}>
        <Typography variant="h6"> Configure Dashboard Data</Typography>
        <Button onClick={openRecentProjects}> Back to recent projects</Button>
      </Box>
      <Box className={classes.dashboardDataContainer}>
        <Box>
          <Typography variant="subtitle2"> Manage Dashboard Dataset</Typography>
        </Box>
        <Box className={classes.dashboardDataHeader}>
          <Typography variant="subtitle2">{`${projectMetadata.name} :: ${selectedDashboardMetadata.name}`}</Typography>
        </Box>
        <Box className={classes.dashboardDataBody}>
          <Typography variant="subtitle2" color="textPrimary">
            Before we can create any visualization, we â€˜ll need some data.
          </Typography>
          <Typography variant="body2">
            Use{' '}
            <Link to={uploadFilePage} component={RouterLink}>
              {' '}
              Upload dataset{' '}
            </Link>
            to add data files to your dashboard.
          </Typography>
        </Box>
      </Box>
    </Box>
  );
}

export default ConfigureDataset;
