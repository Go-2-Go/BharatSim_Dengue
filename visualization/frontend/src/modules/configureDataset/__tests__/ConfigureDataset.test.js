import React from 'react';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import { createMemoryHistory } from 'history';

import ConfigureDataset from '../ConfigureDataset';
import withThemeProvider from '../../../theme/withThemeProvider';
import { ProjectLayoutProvider } from '../../../contexts/projectLayoutContext';

const mockHistoryPush = jest.fn();
const mockHistoryReplace = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
  useHistory: () => ({
    push: mockHistoryPush,
    replace: mockHistoryReplace,
  }),
}));

const history = createMemoryHistory();

const ComponentWithProvider = withThemeProvider(() => (
  <Router history={history}>
    <ProjectLayoutProvider
      value={{
        projectMetadata: { name: 'project1', id: '123' },
        selectedDashboardMetadata: { name: 'dashboard1' },
      }}
    >
      <ConfigureDataset />
    </ProjectLayoutProvider>
  </Router>
));

describe('Configure datasets', () => {
  it('should match snapshot for given dashboard data and project name ', () => {
    const { container } = render(<ComponentWithProvider />);

    expect(container).toMatchSnapshot();
  });

  it('should render configure dataset header with for given dashboard data and project name ', () => {
    const { getByText } = render(<ComponentWithProvider />);

    expect(getByText('project1 :: dashboard1')).toBeInTheDocument();
  });

  it('should navigate to recent projects on click of back to recent button', () => {
    const { getByText } = render(<ComponentWithProvider />);

    fireEvent.click(getByText('Back to recent projects'));

    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });

  it('should navigate to upload data screen on click of upload dataset link', () => {
    const { getByText } = render(<ComponentWithProvider />);

    fireEvent.click(getByText('Upload dataset'));

    expect(history.location.pathname).toEqual('/projects/123/upload-dataset');
  });
});
