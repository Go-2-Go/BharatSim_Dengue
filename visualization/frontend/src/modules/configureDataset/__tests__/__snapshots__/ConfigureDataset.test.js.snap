// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configure datasets should match snapshot for given dashboard data and project name  1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-5"
  >
    <div
      class="MuiBox-root MuiBox-root-6 makeStyles-configureProjectDataBar-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
         Configure Dashboard Data
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
           Back to recent projects
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root MuiBox-root-7 makeStyles-dashboardDataContainer-3"
    >
      <div
        class="MuiBox-root MuiBox-root-8"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
           Manage Dashboard Dataset
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-dashboardDataHeader-2"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          project1 :: dashboard1
        </h6>
      </div>
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-dashboardDataBody-4"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          Before we can create any visualization, we â€˜ll need some data.
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Use
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/projects/123/upload-dataset"
          >
             
            Upload dataset
             
          </a>
          to add data files to your dashboard.
        </p>
      </div>
    </div>
  </div>
</div>
`;
